Preferovat jednodušší způsob jak přidat informace o Pyvu

encukou opened the pull request:
Dobrý den, paní Jechová,

Na stránkách pyvo.cz je teď odkaz, přes který se dá přidat nové Pyvo snadno a rychle. Jen o tom nikdo neví. Mohla byste tuto informaci roznést po pavlačích?

Bohužel se k tomu budete potřebovat naučit brát jiný *argument*, ale doufám, že to zvládnete.

encukou committed:
Advertise an easier way to add meetups.

This needs an API change: we take a slug rather than full URL
to the calendar, so that an URL for the meetup can be generated.
The old way will still work, though.

encukou committed:
Shorten and DRY the chat message source

honzajavorek commented on README.md:
```diff
@@ -12,12 +12,13 @@ Checks whether the next meetup is announced on [pyvo.cz](https://pyvo.cz/). If n
 
-Second argument is the source ics feed for the Pyvo meetup you want to watch. Each meetup tracked on pyvo.cz has the ics feed automatically generated, you can find a link somewhere on the meetup page (e.g. somewhere on [pyvo.cz/praha-pyvo](https://pyvo.cz/praha-pyvo/)).
+Second argument is the meetup slug used in URLs for calendars and meetups, such as `praha-pyvo`.
+(Deprecated: it can also be the URL for the ics feed; the message won't be as nice if you use that.).
```
Nechal bych jen ten nový způsob. Počet přidaných poznámek a ifů podle mě v tomto případě nestojí za zpětnou kompatibilitu. Používá se to zřejmě v jediném souboru, a ten je ještě přímo tady v repozitáři.

honzajavorek commented on jechova.py:
```diff
@@ -17,10 +17,18 @@
     args = parser.parse_args()
 
-    response = requests.get(args.ics_url)
+    if '/' in args.pyvo_slug:
```
Tuto větev bych nadobro odstranil.

encukou replied:
Nechal jsem tam chybovou hlášku, kdyby někdo staré API používal.

honzajavorek commented on jechova.py:
```diff
@@ -41,9 +49,20 @@
+
+            (Existuje i <{docs_url}|podrobnější návod>.)
+        ''').strip())
+    else:
```
Tuto větev bych nadobro odstranil.

honzajavorek commented on jechova.py:
```diff
@@ -41,9 +49,20 @@
+            Pokud už tušíte, o čem Pyvo bude, přidejte to na pyvo.cz — nejlépe
+            přes tlačítko _přidat informace o dalším srazu_ na <{web_url}>.
+
+            (Existuje i <{docs_url}|podrobnější návod>.)
```
Pokud existuje jednodušší návod, nechal bych jen ten. Sekci na docs.pyvec.org bych zachoval pro úplnost, ale tady bych ji už klidně nelinkoval, pokud existuje nějaké nové tlačítko, které to všechno zvládne.

encukou committed:
Only allow slug arguments; streamline message

Slack respects newlines, so textwrap.dedent(), which keeps newlines,
isn't the best tool for the job.

encukou commented:
Tož co takhle?

honzajavorek commented on README.md:
```diff
@@ -12,12 +12,12 @@ Checks whether the next meetup is announced on [pyvo.cz](https://pyvo.cz/). If n
 First argument is a channel name where the message should go. Beware, if you specify it as `#pyvo-praha`, standard shell interprets the `#` as a start of a comment and will ignore the rest of the line. You must pass it either as `pyvo-praha` or as `'#pyvo-praha'` (both works).
 
-Second argument is the source ics feed for the Pyvo meetup you want to watch. Each meetup tracked on pyvo.cz has the ics feed automatically generated, you can find a link somewhere on the meetup page (e.g. somewhere on [pyvo.cz/praha-pyvo](https://pyvo.cz/praha-pyvo/)).
+Second argument is the meetup slug used in URLs for calendars and meetups, such as `praha-pyvo`.
```
Napadlo mě, že toto by mohlo být víc friendly. Kromě insiders podle mě málokdo ví, že na pyvo.cz jsou vůbec nějaké kalendáře, a když už je to teď jen slug, dalo by se to možná napsat takto zjednodušeně:

```suggestion
Second argument is the meetup slug used in the meetup URL. For example, if the meetup URL is `https://pyvo.cz/praha-pyvo/`, the slug would be `praha-pyvo`.
```

honzajavorek commented on jechova.py:
```diff
@@ -41,9 +45,13 @@
+        'Pokud už tušíte, o čem Pyvo bude, přidejte to na pyvo.cz',
+        'přes odkaz _přidat informace o dalším srazu_ na',
+        f'<{web_url}|pyvo.cz/{args.pyvo_slug}>.\n',
+        '(Složitější případy řešte v <#C97Q9HHHT>)',
```
Testoval jsi, jestli to funguje, jo? Myslím ty všelijaké formátovací značky a tak. Budu ti věřit :) Tady bych možná jen za řádek doplnil komentář, že `#C97Q9HHHT` je v lidské mluvě (předpokládám) `#pyvo`:

```suggestion
        '(Složitější případy řešte v <#C97Q9HHHT>)',  # makes a link to the #pyvo channel
```

encukou replied:
Funguje. Naopak `#pyvo` nefunguje :)

honzajavorek reviewed (approved):
👍 udělal jsem ještě nějaké návrhy, ale je to nice to have. Zvaž, co se ti líbí a pokud máš práva, klidně si to pak mergni, pokud ne, mergnu když napíšeš, že je to ready.

encukou committed:
Improve comments/docs

Co-authored-by: Honza Javorek <mail@honzajavorek.cz>

encukou commented:
Upřímně, ta Jechová je strašně vybíravá.

encukou committed:
Shorten a comment to fit on line

encukou commented:
Práva mám, testy se zelenají (i když trochu falešně), tož to začnením.
